### PR TITLE
docs(sinks): Add documentation for encoding.codec

### LINKED ISSUE #8530: `encoding.codec` options do not appear to be rendering on site
Example: https://vector.dev/docs/reference/configuration/sinks/aws_kinesis_firehose/#encoding.codec

Screen Shot 2021-07-30 at 12 26 01 PM

### PR DESCRIPTION
**Pull request by jszwedko**
Fixes #8530

### CODE REVIEW

File: website/cue/reference/components.cue [hunk 1]
```diff
@@ -350,7 +350,7 @@ components: {
 					enabled: bool
 
 					if enabled {
-						default: #EncodingCodec | null
```
Comment (jszwedko) [inline website/cue/reference/components.cue:353]: None of our sinks have a default encoding anymore so I just dropped this.

File: website/cue/reference/components.cue [hunk 2]
```diff
@@ -350,7 +350,7 @@ components: {
...
+						batched: bool | *false
```
Comment (jszwedko) [inline website/cue/reference/components.cue:353]: `encoding.codec` controls the event, and sometimes the batch, encoding so I added this flag to template the descriptions for the encoding differently.

**Review by jszwedko — COMMENTED** [review]
I expect this'll change with Pablo's upcoming encoding work, but this should fix the current documentation.

File: website/cue/reference/components/sinks.cue [hunk 3]
```diff
@@ -163,7 +163,39 @@ components: sinks: [Name=string]: {
 								required:    true
 								type: string: {
 									examples: features.send.encoding.codec.enum
-									syntax:   "literal"
+									let batched = features.send.encoding.codec.batched
+									enum: {
+										for codec in features.send.encoding.codec.enum {
```
Comment (jszwedko) [inline website/cue/reference/components/sinks.cue:168]: I imagine I could refactor this if I had more cue-fu, but this is the best I could come up with.

**Review by spencergilbert — APPROVED** [review]
Gorge' ❤️

**Pull request merged**